70_FLIPFluids/app_resources/descriptor_bindings.hlsl

```diff
@@ -0,0 +1,623 @@
+#ifndef _FLIP_EXAMPLE_BINDINGS_HLSL
+#define _FLIP_EXAMPLE_BINDINGS_HLSL
+
+#include "nbl/builtin/hlsl/cpp_compat.hlsl"
+
+// particlesInit
+NBL_CONSTEXPR uint32_t s_pi = 1;
+NBL_CONSTEXPR uint32_t b_piGridData = 0;
+NBL_CONSTEXPR uint32_t b_piPBuffer = 1;
+
+#ifndef __HLSL_VERSION
+NBL_CONSTEXPR IGPUDescriptorSetLayout::SBinding piParticlesInit_bs1[] = {
+    {
+        .binding = b_piGridData,
+        .type = asset::IDescriptor::E_TYPE::ET_UNIFORM_BUFFER,
+        .createFlags = IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,
+        .stageFlags = asset::IShader::E_SHADER_STAGE::ESS_COMPUTE,
+        .count = 1,
+    },
+    {
+        .binding = b_piPBuffer,
+        .type = asset::IDescriptor::E_TYPE::ET_STORAGE_BUFFER,
+        .createFlags = IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,
+        .stageFlags = asset::IShader::E_SHADER_STAGE::ESS_COMPUTE,
+        .count = 1,
+    }
+};
+#endif
```

Comment:
you don't want this C++ only code in a shared HLSL header, unless somehow that makes the code more readable for you?